### PR TITLE
1.16.2

### PR DESCRIPTION
**Pull request by basaigh**

### CODE REVIEW

File: common/src/main/java/com/github/doctormacc/common/protocols/backwards/v408_to_v407_BackwardsPacketHandler.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        if (AddEntityPacket.class.equals(packet.getClass())) {
+            String entityReplacement = null;
+            if (((AddEntityPacket) packet).getIdentifier().equals("minecraft:piglin_brute")) {
+                entityReplacement = "minecraft:piglin";
+            }
+            if (entityReplacement != null) {
+                ((AddEntityPacket) packet).setIdentifier(entityReplacement);
+            }
+    
+        return true;
```
Comment (rtm516) [inline common/src/main/java/com/github/doctormacc/common/protocols/backwards/v408_to_v407_BackwardsPacketHandler.java:15]: Your missing a bracket before this, creating a build fail.

**Review by rtm516 — CHANGES_REQUESTED** [review]

**Comment by Camotoy** [comment]
Looks good! The v408 class does need to be added to the array in BedrockVersion.java, though.

**Comment by basaigh** [comment]
Oh sorry, thought it done that xd

File: common/src/main/java/com/github/doctormacc/common/protocols/backwards/v408_to_v407_BackwardsPacketHandler.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.github.doctormacc.common.PlayerSession;
+import com.nukkitx.protocol.bedrock.BedrockPacket;
+import com.nukkitx.protocol.bedrock.packet.*;
+
+public class v408_to_v407_BackwardsPacketHandler extends BackwardsPacketHandler {
+
+    @Override
+    public boolean translate(PlayerSession session, BedrockPacket packet, boolean fromUpstream, int translatorIndex) {
+       if (AddEntityPacket.class.equals(packet.getClass())) {
+            String entityReplacement = null;
```
Comment (Camotoy) [inline common/src/main/java/com/github/doctormacc/common/protocols/backwards/v408_to_v407_BackwardsPacketHandler.java]: Couldn't this be simplified to:

```java
            if (((AddEntityPacket) packet).getIdentifier().equals("minecraft:piglin_brute")) {
                ((AddEntityPacket) packet).setIdentifier("minecraft:piglin");
            }
```

**Review by Camotoy — APPROVED** [review]

**Pull request merged**